File: typescript/src/bebop_handlers/reward.ts
import GameState from '../classes/gameState';
import PlayerMediator from '../classes/playerMediator';
import { RewardMsg } from '../models/pactheman.models';
import BebopHandler from './bebopHandler';

export default class RewardMsgHandler implements BebopHandler {
    handleMessage(sender: any, message: any): void {
        message = RewardMsg.decode(message);
        GameState.Instance.gainedRewardAndNewState = message;
        PlayerMediator.completeInitFuture();
        GameState.Instance.signalReward(message);
    }
}

File: dart/lib/bebop_handlers/reward.dart
import 'package:lean_dart_client/classes/game_state.dart';

import 'bebop_handler.dart';
import 'package:lean_dart_client/schemas.g.dart';

class RewardMsgHandler extends BebopHandler {
  @override
  void handleMessage(sender, message) {
    message = RewardMsg.decode(message);
    GameState.instance.gainedRewardAndNewState = message;
    GameState.instance.signalReward(true);
  }
}
